### ISSUE TITLE
TadGAN for personal  multivariate dataset

### ISSUE THREAD

**Issue by 1992AbH** [signals-dev/Orion#231]
* Orion version:
* Python version:python3.7
* Operating System: Google Colab

### Description

I am trying to implement TadGAN for a multivariate dataset (I have 13 columns including the column of the timestamp), I was trying to adapt it for my data but I faced some errors. would you mind explain the steps that I can go through it for modifying the code?
### What I Did
I am using the given example code in colab and I have prepared data into two CSV files; one file contained normal training data and the second is the test set that contains the anomalies.  

  
![training](https://user-images.githubusercontent.com/84004869/117774634-b0f92300-b274-11eb-82b0-d8c4cc8ce2e8.JPG)

![f](https://user-images.githubusercontent.com/84004869/117776408-81e3b100-b276-11eb-9f01-9d78e9ecbdd2.JPG)

![erro](https://user-images.githubusercontent.com/84004869/117775292-544a3800-b275-11eb-8f52-958f95226b86.JPG)

**Comment by sarahmish** [comment]
Hi @1992AbH 

Since you are using your own data, just make sure it follows the same format as we do, which is a `pd.DataFrame` with two columns (`timestamp`, `value`), similar to the one shown here
```
    timestamp     value
0  1222819200 -0.366359
1  1222840800 -0.394108
2  1222862400  0.403625
3  1222884000 -0.362759
4  1222905600 -0.370746
```

Additionally, you need to set the correct hyperparameters for your data.  Based on the frequency of your data you want to change `interval` to reflect that. If your data frequency is 5 minutes then `interval=300` would be most appropriate. The units here are 1 second. 

You can edit hyperparameters as follows
```python3
hyperparameters = {
    "mlprimitives.custom.timeseries_preprocessing.time_segments_aggregate#1": {
        "interval": 300
    }
}

orion = Orion(
    pipeline='tadgan.json',
    hyperparameters=hyperparameters
)
```

**Note**
Our support for multivariate input is limited right now. You will require to change some of the hyperparameters in `orion.primitives.tadgan.TadGAN`, please visit the [primitive page](https://signals-dev.github.io/Orion/user_guides/primitives_pipelines/primitives/TadGAN.html) to find proper documentation there.

**Comment by 1992AbH** [comment]
Hi @sarahmish, Thank you for your information. 
I am still facing the error even with just two columns (data shape :(1000,2)).
I am using Google Colab, and I have just used two columns of the data (timestamp, value). Also, I have changed the interval to "interval": 3600 since the frequency of the data is one hour.
The code as following:

![image](https://user-images.githubusercontent.com/84004869/117924735-20cce380-b331-11eb-9f3d-317a1ddb38b8.png)
the main error 

![image](https://user-images.githubusercontent.com/84004869/117924798-435efc80-b331-11eb-903b-dc9682f1a4ab.png)

Also, I have changed the intervals in the  _tadgan.json_
  
![image](https://user-images.githubusercontent.com/84004869/117925012-9df85880-b331-11eb-8b55-6a0f692b07a7.png)

**Comment by sarahmish** [comment]
Hi @1992AbH, according to the sample data I observe in your input
```
    timestamp     value
0   1           170.721475
1   2           176.461605
2   3           183.871735
3   4           208.256946
```
the delta between each timestamp and another is 1, thus `interval=1` is the appropriate setting.

**Comment by 1992AbH** [comment]
Hello @sarahmish, Thank you so much, it is working.
Now  I am running the code for the data preparation step and learning the model on my own data but I got some error, can you tell me what I can change?.  (Thank you)
  
What I did:
x: is the training data with this shape(1000,1)
![image](https://user-images.githubusercontent.com/84004869/118085867-0f054200-b3fe-11eb-80d5-61d83466a8df.png)
Then the following:
![image](https://user-images.githubusercontent.com/84004869/118086003-3fe57700-b3fe-11eb-9ed0-92988f546d79.png)
![image](https://user-images.githubusercontent.com/84004869/118086100-686d7100-b3fe-11eb-82ef-824cc0361c6a.png)
![image](https://user-images.githubusercontent.com/84004869/118086219-92269800-b3fe-11eb-877a-c17fa71db494.png)
![image](https://user-images.githubusercontent.com/84004869/118086300-b08c9380-b3fe-11eb-9d0f-6f9856dc4705.png)

**Comment by 1992AbH** [comment]
Hi ~~
I am getting this error when I execute the code for univariate data, can you help to solve it. 
![image](https://user-images.githubusercontent.com/84004869/121839960-2180dd00-cd16-11eb-806c-2bc695d3c927.png)
Thank you.

**Comment by 1992AbH** [comment]
Hi ~~
I am getting this error when I execute the code for univariate data, can you help to solve it.
![image](https://user-images.githubusercontent.com/84004869/121840127-905e3600-cd16-11eb-95cf-8aaf32d6a8f2.png)
Thank you.

**Comment by sarahmish** [comment]
hi @1992AbH, I just added a new cell to the notebook to cover some tensorflow/keras issues.
can you please use the updated version, thanks

**Comment by Ashwin-Pokharel** [comment]
@sarahmish which specific notebook are you referring to , as I'm also running into this issue

**Comment by nfpassos** [comment]
Dear @sarahmish , is there any plan to support a multivariate dataset (more than one column for each timestamp like a multimodal).
Thanks in advance.
